site-src/guides/migrating-from-ingress.md

```diff
@@ -0,0 +1,261 @@
+
+The extension points do not include annotations on the Gateway API resources. This approach is strongly discouraged for Gateway implementations.
+
+## Mapping Ingress API features to Gateway API Features
+
+This section will map the Ingress API features to the corresponding Gateway API features. It will cover three major areas:
+
+* Entry points.
+* TLS termination.
+* Routing rules.
+
+### Entry Points
+
+Roughly speaking, an entry point is a combination of an IP address and port through which the data plane is accessible to external clients.
+
+Every Ingress resource has two implicit entry points -- one for HTTP and the other for HTTPS traffic. An Ingress controller provides those entry points. Typically, they will either be shared by all Ingress resources, or every Ingress resource will get dedicated entry points.
+
+In the Gateway API, entry points must be explicitly defined in a [Gateway](/api-types/gateway/) resource. For example, if you want the data plane to handle HTTP traffic on port 80, you need to define a [listener](/references/spec/#gateway.networking.k8s.io/v1beta1.Listener) for that traffic. Typically, a Gateway implementation provides a dedicated data plane for each Gateway resource.
+
+Gateway resources are owned by the cluster operator and the application admin.
+
+### TLS Termination
+
+The Ingress resource supports TLS termination via the [TLS section](https://kubernetes.io/docs/concepts/services-networking/ingress/#tls), where the TLS certificate and key are stored in a Secret.
+
+In the Gateway API, TLS termination is a property of the [listener](/references/spec/#gateway.networking.k8s.io/v1beta1.Listener), and similarly to the Ingress, a TLS certificate and key are also stored in a Secret.
+
+Because the listener is part of the Gateway resource, the cluster operator and application admin own TLS termination.
+
+### Routing Rules
+
+The [path-based routing rules](https://kubernetes.io/docs/concepts/services-networking/ingress/#path-types) of the Ingress resource map directly to the [routing rules](/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteRule) of the [HTTPRoute](/api-types/httproute/).
+
+The [host-header-based routing rules](https://kubernetes.io/docs/concepts/services-networking/ingress/#name-based-virtual-hosting) map to the [hostnames](/references/spec/#gateway.networking.k8s.io/v1beta1.Hostname) of the HTTPRoute. However, note that in the Ingress, each hostname has separate routing rules, while in the HTTPRoute the routing rules apply to all hostnames.
```

Comment:
Hostnames can also be set on Gateway Listeners which should probably be mentioned here as well